WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsConnectedViewModel.swift

```diff
@@ -27,17 +28,56 @@ final class CardReaderSettingsConnectedViewModel: CardReaderSettingsPresentedVie
     var connectedReaderBatteryLevel: String?
     var connectedReaderSoftwareVersion: String?
 
+    /// The connected gateway ID (plugin slug) - useful for the view controller's tracks events
+    var connectedGatewayID: String?
+
     let delayToShowUpdateSuccessMessage: DispatchTimeInterval
 
+    /// The datasource that will be used to help render the related screens
+    ///
+    private(set) lazy var dataSource: CardReaderSettingsDataSource = {
+        return CardReaderSettingsDataSource(siteID: siteID)
+    }()
+
+    var updateType: SoftwareUpdateTypeProperty {
+        optionalReaderUpdateAvailable ? .optional : .required
+    }
+
     init(didChangeShouldShow: ((CardReaderSettingsTriState) -> Void)?,
          knownReaderProvider: CardReaderSettingsKnownReaderProvider? = nil,
          delayToShowUpdateSuccessMessage: DispatchTimeInterval = .seconds(1)) {
         self.didChangeShouldShow = didChangeShouldShow
         self.knownReaderProvider = knownReaderProvider
+        self.siteID = ServiceLocator.stores.sessionManager.defaultStoreID ?? Int64.min
         self.delayToShowUpdateSuccessMessage = delayToShowUpdateSuccessMessage
+
+        configureResultsControllers()
+        loadPaymentGatewayAccounts()
         beginObservation()
     }
 
+    private func configureResultsControllers() {
+        dataSource.configureResultsControllers(onReload: { [weak self] in
+            guard let self = self else {
+                return
+            }
+            self.connectedGatewayID = self.dataSource.cardPresentPaymentGatewayID()
```

Comment:
As per other comment, this doesn't propagate to the ConnectionController, which is responsible for logging some of the events. I don't think that propagating it from here is necessarily the right thing to do though!

Reply:
Fixed in https://github.com/woocommerce/woocommerce-ios/pull/6123/commits/55bcf16e5e7cf62a79aa7ffbca03960e6b387dc9 and https://github.com/woocommerce/woocommerce-ios/pull/6123/commits/7eb5995337c1072590c1422ec8e27b749dc43805